numba/unicode.py

```diff
@@ -1148,6 +1175,126 @@ def impl(a):
     return impl
 
 
+def _is_lower(is_lower, is_upper, is_title):
+    # impl is an approximate translation of:
+    # https://github.com/python/cpython/blob/master/Objects/unicodeobject.c#L11812-L11855
+    # mixed with:
+    # https://github.com/python/cpython/blob/master/Objects/bytes_methods.c#L183-L208
+
+    def impl(data):
+        length = len(data)
+        if length == 1:
+            return is_lower(_get_code_point(data, 0))
+        if length == 0:
+            return False
+
+        cased = False
+        for idx in range(length):
+            code_point = _get_code_point(data, idx)
+            if is_upper(code_point) or is_title(code_point):
+                return False
+            elif not cased and is_lower(code_point):
+                cased = True
+        return cased
+    return impl
+
+
+_always_false = register_jitable(lambda x:False)
+_ascii_is_lower = register_jitable(_is_lower(_Py_ISLOWER, _Py_ISUPPER, _always_false))
```

Comment:
As I understand, this is a method for unicode type - so input data can be unicode only?
Then if yes, no need to handle ascii...
Please correct me if it is wrong

Reply:
Unicode symbols include ASCII ones. So I agree to remove ASCII handling.